oak-run-commons/src/test/java/org/apache/jackrabbit/oak/fixture/DataStoreUtilsTest.java

```diff
@@ -0,0 +1,259 @@
+                        String.format(CONTAINER_DELETED_MESSAGE, CONTAINER_NAME)),
+                Arrays.asList(AUTHENTICATE_VIA_SERVICE_PRINCIPALS_LOG, AUTHENTICATE_VIA_SAS_TOKEN_LOG, AUTHENTICATE_VIA_ACCESS_KEY_LOG),
+                getLogMessages(logAppender));
+        unsubscribe(logAppender);
+    }
+
+    @Test
+    public void delete_non_existing_container_service_principal() throws Exception {
+        final String accountName = getEnvironmentVariable(AZURE_ACCOUNT_NAME);
+        final String clientId = getEnvironmentVariable(AZURE_CLIENT_ID);
+        final String clientSecret = getEnvironmentVariable(AZURE_CLIENT_SECRET);
+        final String tenantId = getEnvironmentVariable(AZURE_TENANT_ID);
+
+        Assume.assumeNotNull(accountName);
+        Assume.assumeNotNull(clientId);
+        Assume.assumeNotNull(clientSecret);
+        Assume.assumeNotNull(tenantId);
+
+        ListAppender<ILoggingEvent> logAppender = subscribeAppender();
+
+        DataStoreUtils.deleteAzureContainer(getConfigMap(null, accountName, null, null, null, clientId, clientSecret, tenantId), CONTAINER_NAME);
+        validate(Arrays.asList(AUTHENTICATE_VIA_SERVICE_PRINCIPALS_LOG, REFRESH_TOKEN_EXECUTOR_SHUTDOWN_LOG, String.format(CONTAINER_DOES_NOT_EXIST_MESSAGE, CONTAINER_NAME)),
+                Arrays.asList(AUTHENTICATE_VIA_AZURE_CONNECTION_STRING_LOG, AUTHENTICATE_VIA_SAS_TOKEN_LOG, AUTHENTICATE_VIA_ACCESS_KEY_LOG),
+                getLogMessages(logAppender));
+        unsubscribe(logAppender);
+    }
+
+
+    @Test
+    public void delete_container_service_principal() throws Exception {
+        final String accountName = getEnvironmentVariable(AZURE_ACCOUNT_NAME);
+        final String clientId = getEnvironmentVariable(AZURE_CLIENT_ID);
+        final String clientSecret = getEnvironmentVariable(AZURE_CLIENT_SECRET);
+        final String tenantId = getEnvironmentVariable(AZURE_TENANT_ID);
+
+        Assume.assumeNotNull(accountName);
+        Assume.assumeNotNull(clientId);
+        Assume.assumeNotNull(clientSecret);
+        Assume.assumeNotNull(tenantId);
+
+        try (AzureBlobContainerProvider azureBlobContainerProvider = AzureBlobContainerProvider.Builder.builder(CONTAINER_NAME)
+                .withAccountName(accountName)
+                .withClientId(clientId)
+                .withClientSecret(clientSecret)
+                .withTenantId(tenantId).build()) {
+            CloudBlobContainer container = azureBlobContainerProvider.getBlobContainer();
+            container.createIfNotExists();
+        }
+
+        ListAppender<ILoggingEvent> logAppender = subscribeAppender();
+
+        DataStoreUtils.deleteAzureContainer(getConfigMap(null, accountName, null, null, null, clientId, clientSecret, tenantId), CONTAINER_NAME);
+        validate(Arrays.asList(AUTHENTICATE_VIA_SERVICE_PRINCIPALS_LOG, REFRESH_TOKEN_EXECUTOR_SHUTDOWN_LOG,
```

Comment:
Can we also validate on container existing or not ? Just validating logs is not sufficient.

Reply:
done, added `assertFalse(container.exists());` after deleting the container